### PR TITLE
Use `for-of` loop instead of a `for` loop

### PR DESCRIPTION
**Pull request by apupier**
as recommended by Sonar
https://sonarcloud.io/project/issues?id=vscode-camelk&open=AXe5MjsIomOWeNW5TYmX&resolved=false&rules=typescript%3AS4138&types=CODE_SMELL

Signed-off-by: Aurélien Pupier <apupier@redhat.com>

### CODE REVIEW

File: src/CamelKNodeProvider.ts [hunk 1]
```diff
@@ -150,13 +150,13 @@ export class CamelKNodeProvider implements vscode.TreeDataProvider<TreeNode> {
 		if (output) {
 			const lines: Array<string> = output.split('\n');
 			for (let entry of lines) {
-				const line: Array<string> = entry.split('  ');
+				const spaceSplittedLine: Array<string> = entry.split('  ');
 				const cleanLine: Array<string> = new Array<string>();
-				for (var i=0; i < line.length; i++) {
-					if (line[i].trim().length === 0) {
+				for (let integration of spaceSplittedLine) {
```
Comment (bfitzpat) [inline src/CamelKNodeProvider.ts:155]: Does entry.split always return a value? Or do we need to watch for undefined/empty array here?

Comment (apupier) [reply src/CamelKNodeProvider.ts:155]: it always returns an array:

```
    /**
     * Split a string into substrings using the specified separator and return them as an array.
     * @param splitter An object that can split a string.
     * @param limit A value used to limit the number of elements returned in the array.
     */
    split(splitter: { [Symbol.split](string: string, limit?: number): string[]; }, limit?: number): string[];
```

Comment (bfitzpat) [reply src/CamelKNodeProvider.ts:155]: so the `for-of` construct doesn't execute if the array is empty?

Comment (apupier) [reply src/CamelKNodeProvider.ts:155]: yes, if the array is empty, it is not entering the statement block.

**Review by bfitzpat — APPROVED** [review]

**Pull request merged**